open-craft/problem-builder
# Step Builder: New block type: "Plot"

This PR adds a "plot" block to Step Builder that can be used to visualize answers to pairs of rating questions.

**Testing**

1. Add Step Builder to a unit.
2. Add two Mentoring Step blocks and a Review Step block to Step Builder.
3. Add two Rating questions to the first Mentoring Step block.
4. Add a Plot block to the second Mentoring Step block.
5. For each of the Rating questions, grab the corresponding "Question ID" from the EDIT dialog in Studio (last field, underneath "Label for high ratings").
6. Bring up the EDIT dialog for the Plot block and add the following to the "Claims and associated questions" field:

        This is an important claim, 7cf9575c53d5abc43fddedc4b06ae1a835dd4bf3, 79020e44630e246077a2bce92f45562fe24769cd

   Make sure to replace the two hashes with the question IDs you grabbed in the previous step.
7. Publish the unit.
8. Complete the first step of the unit by submitting answers to the rating questions.
9. Navigate to the second step. The plot should show the answers you gave in the form of a single point.
10. Repeat the previous two steps as another user and make sure to choose different answers for the rating questions. When you get to the second step, the point representing the average value of all submissions for the rating questions should be located in a different place than the point representing your own answer.
